generator/main.go

```diff
@@ -65,10 +74,30 @@ func recordWritesErrored(count float64) {
 	writesErrored.Add(count)
 }
 
+func recordPatchesCompleted(count float64) {
+	patchesCompleted.Add(count)
+}
+
+func recordPatchesErrored(count float64) {
+	patchesErrored.Add(count)
+}
+
+const defaultRocksetEndpoint = "https://api.rs2.usw2.rockset.com"
+
+var max_doc_id int
+
 func main() {
 	wps := mustGetEnvInt("WPS")
 	batchSize := mustGetEnvInt("BATCH_SIZE")
 	destination := mustGetEnvString("DESTINATION")
+	num_docs := getEnvDefaultInt("NUM_DOCS", -1)
+	do_patch := getEnvDefaultBool("PATCH", false)
+	patch_mode := getEnvDefault("PATCH_MODE", "replace")
+
+	if !(patch_mode == "replace" || patch_mode == "add") {
+		panic("Invalid patch mode specified")
+	}
+	pps := getEnvDefaultInt("PPS", wps)
```

Comment:
should we use WPS for this too? not strong about it though

Reply:
If it isn't specified, wps will be used. I think it is fine.